cfme/tests/cloud/test_providers.py

```diff
@@ -1297,3 +1300,65 @@ def test_add_ec2_provider_with_non_default_url_endpoint():
             2. Refresh should complete without errors
     """
     pass
+
+
+@pytest.mark.ignore_stream("5.10")
+def test_add_ec2_provider_with_sts_assume_role(appliance, request):
+    """
+    Requires:
+        The requirement is only on EC2 side and needs to be added manually once.
+        1. Role which has all the required permissions to manage CFME
+        2. Edit Trust relationship policy for this role to:
+            {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Effect": "Allow",
+                  "Principal": {
+                    "AWS": "arn:aws:iam::NNNNNNNNNNNN:root"
+                  },
+                  "Action": "sts:AssumeRole"
+                }
+              ]
+            }
+        3. Have policy with AssumeRole permission:
+            {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": "sts:AssumeRole",
+                        "Resource": "arn:aws:iam::NNNNNNNNNNNN:role/RoleForCFME"
+                    }
+                ]
+            }
+        4. Have an user with only attached policy created in last step
+
+    Polarion:
+        assignee: mmojzis
+        casecomponent: Cloud
+        initialEstimate: 1/2h
+        caseimportance: high
+        casecomponent: Cloud
+        testSteps:
+            1. Go to Compute -> Cloud -> Providers
+            2. Add EC2 Provider with these fields filled in:
+        expectedResults:
+            1.
+            2. Provider should be successfully added.
+    """
+    assume_role_creds = cfme_data['management_systems']['ec2west']['sts_assume_role']['credentials']
+    creds = Credential(principal=credentials[assume_role_creds]['username'],
+                       secret=credentials[assume_role_creds]['password'])
+    endpoint = EC2Endpoint(
+        assume_role_arn=cfme_data['management_systems']['ec2west']['sts_assume_role']['role_arn'],
+        credentials=creds)
+
+    collection = appliance.collections.cloud_providers
+    prov = collection.instantiate(
+        prov_class=EC2Provider, name=fauxfactory.gen_alphanumeric(5),
+        region_name=cfme_data['management_systems']['ec2west']['region_name'],
+        endpoints=endpoint)
+
+    request.addfinalizer(prov.delete_if_exists)
+    prov.create()
```

Comment:
Is there any verification that needs to be done after the provider is successfully added? Will the test error out or fail if the  provider is not successfully added?

Reply:
Well I wanted to add some smoke tests after adding this provider, but I think for now checking whether provider refresh was successful is sufficient.